### ISSUE TITLE
Error `object.to_json` with `date` or `timespan`

### ISSUE THREAD

**Issue by r-Larch** [scriban/scriban#555]
The following code produces an error:

```f#
{{ "1024-06-26" | date.parse | object.to_json }}
// -> error : CurrentDepth (1000) is equal to or larger than the maximum allowed depth of 1000. 
// -> Cannot write the next JSON object or array.
// Expected Result: "1024-06-26T00:00:00+02:00"
```

The error comes from `DateTime` being interpreted as 'object' and not as a value.

## A possible fix 

This and probably all such cases could be fixed with the following change:
- Ensure that all objects implementing `IFormattable` are being threaded as a value rather than an object.

https://github.com/scriban/scriban/blob/38df14946721238d2f4badc7c0b6e8a1c594e23c/src/Scriban/Functions/ObjectFunctions.cs#L481-L532

```diff
var type = value?.GetType() ?? typeof(object);
if (
    value is null ||
    value is string ||
    value is bool ||
-    type.IsPrimitiveOrDecimal()
+    type.IsPrimitiveOrDecimal() ||
+    value is IFormattable
)
{
    JsonSerializer.Serialize(writer, value, type);
}
else if (value is IList || type.IsArray) {
```

**Comment by xoofx** [comment]
Yep, PR welcome with a test.

**Comment by r-Larch** [comment]
Great, I i created the issue to remember, PR follows wenn I have a little more time 👌

**Comment by ChopperRob** [comment]
Hi @r-Larch , i see you had an similar issue as we did, i just had a pr merged fixing the DateTime issue, #558 .  It does not handle all IFormattable, but only the DateTime which was enough for us.
